test/extended/images/helper.go

```diff
@@ -56,3 +80,277 @@ func CheckPageContains(oc *exutil.CLI, endpoint, path, contents string) (bool, e
 	}
 	return strings.Contains(response, contents), nil
 }
+
+// BuildAndPushImageOfSizeWithBuilder tries to build an image of wanted size and number of layers. Built image
+// is stored as an image stream tag <name>:<tag>. If shouldSucceed is false, a build is expected to fail with
+// a denied error. Note the size is only approximate. Resulting image size will be different depending on used
+// compression algorithm and metadata overhead.
+func BuildAndPushImageOfSizeWithBuilder(
+	oc *exutil.CLI,
+	dClient *dockerclient.Client,
+	namespace, name, tag string,
+	size uint64,
+	numberOfLayers int,
+	shouldSucceed bool,
+) error {
+	istName := name
+	if tag != "" {
+		istName += ":" + tag
+	}
+
+	bc, err := oc.REST().BuildConfigs(namespace).Get(name)
+	if err == nil {
+		if bc.Spec.BuildSpec.Output.To.Kind != "ImageStreamTag" {
+			return fmt.Errorf("Unexpected kind of buildspec's output (%s != %s)", bc.Spec.BuildSpec.Output.To.Kind, "ImageStreamTag")
+		}
+		bc.Spec.BuildSpec.Output.To.Name = istName
+		if _, err = oc.REST().BuildConfigs(namespace).Update(bc); err != nil {
+			return err
+		}
+	} else {
+		err = oc.Run("new-build").Args("--binary", "--name", name, "--to", istName).Execute()
+		if err != nil {
+			return err
+		}
+	}
+
+	tempDir, err := ioutil.TempDir("", "name-build")
+	if err != nil {
+		return err
+	}
+
+	dataSize := calculateRoughDataSize(oc.Stdout(), size, numberOfLayers)
+
+	lines := make([]string, numberOfLayers+1)
+	lines[0] = "FROM scratch"
+	for i := 1; i <= numberOfLayers; i++ {
+		blobName := fmt.Sprintf("data%d", i)
+		if err = createRandomBlob(path.Join(tempDir, blobName), dataSize); err != nil {
+			return err
+		}
+		lines[i] = fmt.Sprintf("COPY %s /%s", blobName, blobName)
+	}
+	if err = ioutil.WriteFile(path.Join(tempDir, "Dockerfile"), []byte(strings.Join(lines, "\n")+"\n"), 0644); err != nil {
```

Comment:
Replacing in all the occurences (if possible).
On 05/26/2016 11:05 AM, Maciej Szulik wrote:
>
> In test/extended/images/helper.go <https://github.com/openshift/origin/pull/8195#discussion_r64713231>:
>
> > +	if err != nil {
> > +		return err
> > +	}
> > +
> > +	dataSize := calculateRoughDataSize(oc.Stdout(), size, numberOfLayers)
> > +
> > +	lines := make([]string, numberOfLayers+1)
> > +	lines[0] = "FROM scratch"
> > +	for i := 1; i <= numberOfLayers; i++ {
> > +		blobName := fmt.Sprintf("data%d", i)
> > +		if err = createRandomBlob(path.Join(tempDir, blobName), dataSize); err != nil {
> > +			return err
> > +		}
> > +		lines[i] = fmt.Sprintf("COPY %s /%s", blobName, blobName)
> > +	}
> > +	if err = ioutil.WriteFile(path.Join(tempDir, "Dockerfile"), []byte(strings.Join(lines, "\n")+"\n"), 0644); err != nil {
>
> Whenever using short-if use |:=|
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub 
> <https://github.com/openshift/origin/pull/8195/files/0f18bf53a94fa198a09aa1504510eb66dee28a0d..2cbc04f597c0bc9374d95a8d6e718c8b31e9a7a0#r64713231>
>